src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -24,6 +24,18 @@ namespace OpenTelemetry.Metrics;
 /// </summary>
 public struct MetricPoint
 {
+    // Represents the number of update threads using this MetricPoint at any given point of time.
+    // If the value is equal to int.MinValue which is -2147483648, it means that this MetricPoint is available for reuse.
+    // We never increment the ReferenceCount for MetricPoint with no tags (index == 0) and the MetricPoint for overflow attribute,
+    // but we always decrement it (in the Update methods). This should be fine.
+    // ReferenceCount doesn't matter for MetricPoint with no tags and overflow attribute as they are never reclaimed.
+    internal int ReferenceCount;
```

Comment:
@alanwest @vishweshbankwar This is going to add to our memory consumption (4 bytes x MaxMetricPoints). Below there is another field `AggregationType aggType` which I'm guessing is also taking up 4 bytes. I don't think either of those two things really need the full 4 bytes, what if we did some bit manipulation on a single `int` field to process both needs with better memory utilization?